fvm/src/machine/engine.rs

```diff
@@ -125,26 +128,67 @@ impl Engine {
                     &cid.to_string()
                 )
             })?;
-            let module = Module::from_binary(&self.0.engine, wasm.as_slice())?;
+            let module = self.load_raw(wasm.as_slice(), mctx)?;
             cache.insert(*cid, module);
         }
         Ok(())
     }
 
     /// Load some wasm code into the engine.
-    pub fn load_bytecode(&self, k: &Cid, wasm: &[u8]) -> anyhow::Result<Module> {
+    pub fn load_bytecode(
+        &self,
+        k: &Cid,
+        wasm: &[u8],
+        mctx: &MachineContext,
```

Comment:
So, this engine was supposed to be independent of machine, epoch, pricelist, etc. We're probably going to need to move to a model where we construct a `NetworkConfig` (contains the network version, pricelist, etc.).